crates/arroyo-planner/src/tables.rs

```diff
@@ -511,16 +534,55 @@ impl Table {
                     Field::new(name, data_type, nullable),
                 );
 
-                let generating_expression = column.options.iter().find_map(|option| {
-                    if let ColumnOption::Generated {
-                        generation_expr, ..
-                    } = &option.option
-                    {
-                        generation_expr.clone()
-                    } else {
-                        None
-                    }
-                });
+                let generating_expression =
+                    column.options.iter().find_map(|option| {
+                        if let ColumnOption::Generated {
+                            generation_expr, ..
+                        } = &option.option
+                        {
+                            if let Some(sqlparser::ast::Expr::Function(
+                                sqlparser::ast::Function { name, args, .. },
+                            )) = generation_expr
+                            {
+                                if name
+                                    .0
+                                    .iter()
+                                    .any(|ident| ident.value.to_lowercase() == "metadata")
+                                {
+                                    match args {
+                                        FunctionArguments::List(arg_list) => {
+                                            match arg_list.args.first() {
+                                            Some(FunctionArg::Unnamed(
+                                                sqlparser::ast::FunctionArgExpr::Expr(
+                                                    sqlparser::ast::Expr::Value(
+                                                        sqlparser::ast::Value::SingleQuotedString(
+                                                            value,
+                                                        ),
+                                                    ),
+                                                ),
+                                            )) => {
+                                                connector_metadata_columns.insert(
+                                                    column.name.value.to_string(),
+                                                    value.to_string(),
+                                                );
+                                            }
+                                            _ => {
+                                                "Unsupported argument format.".to_string();
```

Comment:
I'm a bit confusing by the intent here—this string is created by never used. Is this meant to return an error?